src/internal.cc

```diff
@@ -175,6 +175,9 @@ internalHostnameIs(const char *arg)
         if (0 == strcmp(arg, w->key))
             return 1;
 
+    if (0 == strcmp(arg, "localhost"))
+        return 1;
+
     return 0;
```

Comment:
> PR description: Presence of cache_object:// scheme alone used to be enough to determine the use of an internal request, so the checks of hostname did not need to care about 'localhost' being sent instead of the actual assigned hostname. Now that the default squidclient URL is http://localhost:3128/ this additional possible hostname matters.

Since Squid supported non-cache_object internal requests before that scheme was removed, I do not think we can claim that hostname checks were not needed before the removal and are needed after the removal. AFAICT, those checks were needed before and are still needed now to identify internal URLs matching the current Squid instance.

I need more time to suggest a specific fix for the PR description.

Reply:
> I need more time to suggest a specific fix for the PR description.

Done: I updated PR description text accordingly. Please adjust further as needed.

I consider this change request resolved.